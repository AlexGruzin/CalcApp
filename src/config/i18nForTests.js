import i18n from 'i18next';

// locales
import US from '../locales/en';

i18n.init({
  debug: false,
  fallbackLng: ['US'],
  lng: 'US',

  ns: ['auth', 'resetPassword'],
  defaultNS: 'auth',

  resources: {
    US,
  },
});

export default i18n;

