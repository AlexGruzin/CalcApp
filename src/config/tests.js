import React from 'react';
import { render } from '@testing-library/react';
import { I18nextProvider } from 'react-i18next'
import i18n from "./i18nForTests";
import jsdom from 'jsdom';
const { JSDOM } = jsdom;

/*
 * jest-dom
 * https://github.com/testing-library/jest-dom
 */
import '@testing-library/jest-dom/extend-expect';

/*
 * jest-fetch-mock
 * https://github.com/jefflau/jest-fetch-mock
 */
require('jest-fetch-mock').enableMocks();

if (global.document) {
  global.document.createRange = () => ({
    setStart: () => {
    },
    setEnd: () => {
    },
    commonAncestorContainer: {
      nodeName: 'BODY',
      ownerDocument: document,
    },
    createContextualFragment: str => JSDOM.fragment(str),
    // createContextualFragment: str => document.createElement('hr'),
  });
  global.open = () => undefined;
}

export const renderWithi18next = Component => {
  const Comp = React.cloneElement(Component, {
    changeLanguage: lng => {
      i18n.changeLanguage(lng);
      rerender(<I18nextProvider i18n={i18n}>{Comp}</I18nextProvider>);
    }
  });
  const defaultRender = render(
    <I18nextProvider i18n={i18n}>{Component}</I18nextProvider>
  );
  const { rerender } = defaultRender;
  return defaultRender;
};


// /**
//  * Wraps a component with an Intl Provider.
//  * @param {Object} component The component to render
//  */
// const withTestIntl = component => <IntlProvider locale="en">{component}</IntlProvider>;
//
// /*
//  * https://testing-library.com/docs/example-react-intl
//  */
// export const renderWithReactIntl = component => render(withTestIntl(component));

