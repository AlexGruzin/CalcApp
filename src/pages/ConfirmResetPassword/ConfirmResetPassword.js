import React, { useState, useCallback } from 'react';
import PropTypes from 'prop-types';
import { useForm } from 'react-hook-form';
import { passwordValidation } from 'helpers/validation/auth';
import { useTranslation } from 'react-i18next';
import Button from 'components/Button';
import CustomInput from 'components/CustomInput';

import './ConfirmResetPassword.styles.less';

const ConfirmResetPassword = ({ confirmPassword }) => {
  const { t } = useTranslation();
  const { register, handleSubmit: handleSubmitForm, errors } = useForm({ mode: 'onChange', shouldUnregister: true });
  const [password, setPassword] = useState({});

  const handleSubmit = () => {
    confirmPassword(password);
  };

  const onChangeField = useCallback(e => {
    const { value } = e.target;
    setPassword(value);
  }, []);

  const validationErrorMsg = errors[Object.keys(errors)[0]]?.message;

  return (
    <div className="confirm-psw">
      <img className="confirm-psw__logo" src="/svg/logo.svg" alt="Calc logo" />
      <div className="confirm-psw__content">
        <form onSubmit={handleSubmitForm(handleSubmit)} name="confirm-psw-form" className="confirm-psw__form">
          <CustomInput
            onChange={onChangeField}
            type="password"
            name="password"
            ref={register(passwordValidation())}
            className="confirm-psw__password"
            placeholder={t('confirm:reset.password')}
            isError={!!validationErrorMsg}
            errMessage={validationErrorMsg}
            required
          />
          <Button
            disabled={!!validationErrorMsg}
            className="confirm-psw__submit"
            type="submit"
            title={t('confirm:reset.confirm')}
          />
        </form>
      </div>
    </div>
  );
};

ConfirmResetPassword.propTypes = {
  confirmPassword: PropTypes.func.isRequired,
};

export default ConfirmResetPassword;
