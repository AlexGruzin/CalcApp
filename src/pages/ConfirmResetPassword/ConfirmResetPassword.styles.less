@import "~styles/generic.less";

.confirm-psw {
  display: flex;
  flex-direction: column;
  height: 100%;
  width: 100%;
  justify-content: center;
  align-items: center;
}

.confirm-psw__logo {
  .calc-logo();
}

.confirm-psw__content {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 100%;
  margin-bottom: 100px;
}

.confirm-psw__form {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: var(--space-x2);
  width: 590px;
  min-width: 590px;
  max-width: 590px;
  height: 397px;
  max-height: 397px;
  min-height: 397px;
  background-color: var(--white);
  border-radius: 13px;
}

.confirm-psw__password input {
  height: 36px;
  width: 294px;
  text-align: center;
  border-radius: 30px;
  border: 1px solid var(--border-grey);
}

.confirm-psw__password {
  margin-bottom: 12px;
}

.confirm-psw__submit {
  height: 35px;
  min-height: 35px;
  width: 162px;
  border-radius: 18px;
  background-color: var(--button-purple);
  text-transform: capitalize;
  margin-bottom: var(--space-x3);
}
