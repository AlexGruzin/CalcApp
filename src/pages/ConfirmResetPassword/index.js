import { connect } from 'react-redux';

import { initConfirmPassword } from 'sagas/actions/confirmResetPassword';
import ConfirmResetPassword from './ConfirmResetPassword';

export default connect(null, { confirmPassword: initConfirmPassword })(ConfirmResetPassword);
