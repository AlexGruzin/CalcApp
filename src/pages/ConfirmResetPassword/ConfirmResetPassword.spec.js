import React from 'react';
import { fireEvent, waitFor } from '@testing-library/react';
import { renderWithi18next } from 'config/tests';
import userEvent from '@testing-library/user-event';
import ConfirmResetPassword from './ConfirmResetPassword';

describe('ConfirmResetPassword tests', () => {
  it('Check snapshots of ConfirmResetPassword are equal.', () => {
    const { asFragment } = renderWithi18next(<ConfirmResetPassword confirmPassword={() => undefined} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Press submit to send api request with valid password.', async () => {
    const mockSubmit = jest.fn();
    const { getAllByRole, getByLabelText } = renderWithi18next(<ConfirmResetPassword confirmPassword={mockSubmit} />);
    const btns = getAllByRole('button');
    const input = getByLabelText('password');

    fireEvent.change(input, { target: { value: '12345678' } }); // enter new password

    userEvent.click(btns[0]);
    waitFor(() => {
      expect(mockSubmit).toBeCalledTimes(1);
    });
  });

  it('Press submit to send api request with invalid password.', async () => {
    const mockSubmit = jest.fn();
    const { getAllByRole, getByLabelText } = renderWithi18next(<ConfirmResetPassword confirmPassword={mockSubmit} />);
    const btns = getAllByRole('button');
    const input = getByLabelText('password');

    fireEvent.change(input, { target: { value: '1' } }); // enter new password

    userEvent.click(btns[0]);

    waitFor(() => {
      expect(mockSubmit).toBeCalledTimes(0);
    });
  });
});
