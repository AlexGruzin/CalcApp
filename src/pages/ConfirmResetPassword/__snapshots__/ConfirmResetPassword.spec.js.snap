// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmResetPassword tests Check snapshots of ConfirmResetPassword are equal. 1`] = `
<DocumentFragment>
  <div
    class="confirm-psw"
  >
    <img
      alt="Calc logo"
      class="confirm-psw__logo"
      src="/svg/logo.svg"
    />
    <div
      class="confirm-psw__content"
    >
      <form
        class="confirm-psw__form"
        name="confirm-psw-form"
      >
        <div
          class="confirm-psw__password custom-input"
        >
          <input
            aria-label="password"
            id="password"
            name="password"
            placeholder="New password"
            required=""
            type="password"
          />
          <div
            class="custom-input__error"
          />
        </div>
        <button
          class="custom-button confirm-psw__submit"
          type="submit"
          value=""
        >
          Confirm
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
